lib/msal-common/src/client/RefreshTokenClient.ts

```diff
@@ -59,7 +61,27 @@ export class RefreshTokenClient extends BaseClient {
             undefined,
             true,
             request.forceCache
-        );
+        ).then((result: AuthenticationResult) => {
+            let refreshTokenSize;
+            if (response.body?.refresh_token) {
+                refreshTokenSize = response.body.refresh_token.length || 0;
+            }
+            atsMeasurement?.endMeasurement({
+                success: true,
+                refreshTokenSize: refreshTokenSize
+            });
+            return result;
```

Comment:
If you are wanting to differentiate, you need to explicitly check the type of `body.refresh_token`. Simply doing a truthy check will result in `undefined` and a string of length 0 being treated the same (which you indicated you don't want to do). Furthermore, this can be simplified (also, please leave comment detailing why this is here, as it is not necessarily obvious):

```suggestion
            // Don't set refreshtokensize if no refresh token
            const refreshTokenSize = typeof response.body?.refresh_token === "undefined" ? undefined : (response.body.refresh_token.length || 0);
            atsMeasurement?.endMeasurement({
                success: true,
                refreshTokenSize: 
            });
            return result;
```

Reply:
Please add a unit test to confirm this logic.

Reply:
But as noted before, I would personally not to differentiate between undefined and length of zero:

```suggestion
            atsMeasurement?.endMeasurement({
                success: true,
                refreshTokenSize: response.body.refresh_token?.length || 0
            });
            return result;
```

Reply:
Thanks for your helpful feedback  @jasonnutter ! I have included the code changes based on the second approach(i.e not semantically differentiating between RT of size 0 and undefined). Have included appropriate tests in the RefreshTokenClient. Thanks!